contrib/TestHarness2/test_harness/run.py

```diff
@@ -73,8 +73,14 @@ def __init__(self, test_dir: Path, binaries: OrderedDict[Version, Path]):
         self.rare_priority: int = int(os.getenv("RARE_PRIORITY", 10))
 
         for subdir in self.test_dir.iterdir():
-            if subdir.is_dir() and subdir.name in config.test_dirs:
+            if subdir.is_dir() and subdir.name in config.test_types_to_run:
                 self.walk_test_dir(subdir)
+        
+        # Also check for test files directly in the test source directory
+        # This handles cases where tarball contains test files without subdirectory structure
+        for item in self.test_dir.iterdir():
+            if item.is_file() and (item.suffix == ".txt" or item.suffix == ".toml"):
+                self.parse_txt(item)
```

Comment:
Historically, we don't run Joshua tests for these test files. Some of them may not work in simulation. Is this going to change the behavior, i.e., running them now?